helix-view/src/tree.rs

```diff
@@ -342,16 +342,35 @@ impl Tree {
 
         // take the area
         // fetch the node
-        // a) node is view, give it whole area
+        // a) node is view
+        //    1) view is centered and the parent container is not a vertical layout or
+        //       with only 1 child (the current view), center the view
+        //    2) give the whole area
         // b) node is container, calculate areas for each child and push them on the stack
 
         while let Some((key, area)) = self.stack.pop() {
-            let node = &mut self.nodes[key];
+            let parent = self.nodes[key].parent;
 
-            match &mut node.content {
+            // Parent must always be a container
+            let (parent_child_count, parent_layout) = match &self.nodes[parent].content {
+                Content::Container(container) => (container.children.len(), container.layout),
+                Content::View(_) => unreachable!(),
+            };
+
+            match &mut self.nodes[key].content {
                 Content::View(view) => {
                     // debug!!("setting view area {:?}", area);
-                    view.area = area;
+                    if view.is_centered
+                        && (parent_child_count <= 1 || parent_layout != Layout::Vertical)
+                    {
+                        let width = std::cmp::min(std::cmp::max(area.width / 2, 120), area.width);
+                        let area =
+                            Rect::new(area.width / 2 - width / 2, area.y, width, area.height);
+
+                        view.area = area;
+                    } else {
+                        view.area = area;
+                    }
```

Comment:
This gives the view the entire area but doesn't affect any of the other views. This means that you're relying on view ordering (this view should be the last one rendered) otherwise other views might get rendered over it. We also pay the performance cost of rendering all other views even though they're not on screen. (Also what happens if I change focus to a different view but this one is still centered?)

I'd rather see this done via a `zoom` bool flag on the tree. If set to `true` then the editor should just render the currently focused node and ignore the rest of the tree. The reason we haven't done this yet is because this will likely impact a bunch of other viewport calculations.

Reply:
Hi, thanks for the feedback, but... to be honest I made this PR 7 months ago, so I don't remember at all why I made things in this way, and I don't have sufficent spare time to invest in this PR anymore :/ if someone is motivated enough to take over my work (or redoing it from scratch ^^') it would be better